block/block.py

```diff
@@ -26,14 +27,14 @@ def __init__(self, previous_hash=None, transactions=None):
         # define block header attributes
         self.version = version.encode('utf8')  # 4 bytes
         self.previous_hash = previous_hash.encode('utf8')  # 32 bytes
-        self.merkle_root = self.merkle_root()  # 32 bytes
+        self.merkle_root = self.merkle_root(transactions)  # 32 bytes

```

Comment:
why do we need to pass transactions to the merkle_root function if the transactions themselves are stored in the block object and accessible under self?

Reply:
Oh I see... Ok coolio